File: python/55a_822A_ImBoredWithLife.py
def fact(a):
    cnt = 1
    for i in range(2, a+1):
        cnt *= i
    return cnt

def solve(a, b):
    return fact(min(a, b))

def main():
    a, b = input().split()
    a, b = int(a), int(b)
    print(solve(a, b))

if __name__ == "__main__":
    main()

File: cpp/55a_822A_ImBoredWithLife.cpp
#include<bits/stdc++.h>

int fact(int a){
    int cnt = 1;
    for(int i = 1; i <= a; i++){
        cnt *= i;
    }
    return cnt;
}

int solve(int a, int b){
    return fact(std::min(a, b));
}

int main(){
    int a, b;
    std::cin>>a>>b;
    std::cout<<solve(a, b);
    return 0;
}